debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/transforms/partitions/ComputePartition.java

```diff
@@ -0,0 +1,142 @@
+            int partition = computePartition(fieldValue, table);
+
+            LOGGER.debug("Message {} will be sent to partition {}", envelope, partition);
+
+            return r.newRecord(r.topic(), partition,
+                    r.keySchema(),
+                    r.key(),
+                    r.valueSchema(),
+                    envelope,
+                    r.timestamp());
+        }
+        catch (Exception e) {
+            LOGGER.error("Error occurred while processing message {}. Skipping SMT", envelope);
+            return r;
+        }
+    }
+
+    private boolean skipRecord(String table) {
+
+        if (!tableNames.contains(table)) {
+            LOGGER.debug("Table {} is not configured. Skipping STM", table);
+        }
+        if (!fieldNameByTable.containsKey(table)) {
+            LOGGER.debug("No field name property defined for table {}. Skipping STM", table);
+        }
+        if (!numberOfPartitionsByTable.containsKey(table)) {
+            LOGGER.debug("No number of partition property defined for table {}. Skipping STM", table);
+        }
+
+        return !tableNames.contains(table) || !fieldNameByTable.containsKey(table) || !numberOfPartitionsByTable.containsKey(table);
+    }
+
+    private String getTableName(Struct envelope) {
+
+        Struct struct = (Struct) envelope.get(SOURCE);
+        return struct.getString(TABLE_FIELD_NAME);
+    }
+
+    private int computePartition(Object fieldValue, String table) {
+
+        return fieldValue.hashCode() % numberOfPartitionsByTable.get(table);
+    }
+
+    private Struct extractPayload(Struct envelope) {
+
+        String operation = envelope.getString(OPERATION);
+
+        Struct struct = (Struct) envelope.get(AFTER);
+        if (Envelope.Operation.DELETE.code().equals(operation)) {
+            struct = (Struct) envelope.get(BEFORE);
+        }
+        return struct;
+    }
+
+    @Override
+    public ConfigDef config() {
+
+        ConfigDef config = new ConfigDef();
+        // group does not manage validator definition. Validation will not work here.
+        return Field.group(config, "partitions",
+                PARTITION_TABLE_FIELD_NAME_MAPPINGS_FIELD, FIELD_TABLE_PARTITION_NUM_MAPPINGS_FIELD, PARTITION_TABLE_LIST_FIELD);
+    }
+
+    @Override
+    public void configure(Map<String, ?> props) {
+
+        final Configuration config = Configuration.from(props);
+        smtManager = new SmtManager<>(config);
+        smtManager.validate(config, Field.setOf(PARTITION_TABLE_FIELD_NAME_MAPPINGS_FIELD));
+
+        tableNames = config.getStrings(PARTITION_TABLE_LIST_FIELD, LIST_SEPARATOR);
+        numberOfPartitionsByTable = ComputePartitionConf.parseIntMappings(config.getStrings(FIELD_TABLE_PARTITION_NUM_MAPPINGS_FIELD, LIST_SEPARATOR));
+        fieldNameByTable = ComputePartitionConf.parseMappings(config.getStrings(PARTITION_TABLE_FIELD_NAME_MAPPINGS_FIELD, LIST_SEPARATOR));
```

Comment:
I thik it would be good to have a validation here checking if numberOfPartitionsByTable and fieldNameByTable are consistent.